autode/hessians.py

```diff
@@ -213,19 +200,21 @@ def _proj_matrix(self) -> np.ndarray:
 
         # Construct M^1/2, which as it's diagonal, is just the roots of the
         # diagonal elements
-        masses = np.repeat(
-            [atom.mass for atom in self.atoms], repeats=3, axis=np.newaxis
-        )
-        m_half = np.diag(np.sqrt(masses))
+        masses = np.repeat([atom.mass for atom in self.atoms], repeats=3)
+        m_half = np.sqrt(masses)
 
         for t_i in (t1, t2, t3, t4, t5, t6):
-            t_i[:] = np.dot(m_half, np.array(t_i))
-            t_i /= np.linalg.norm(t_i)
+            t_i *= m_half
 
         # Generate a transform matrix D with the first columns as translation/
         # rotation vectors with the remainder as random orthogonal columns
-        M = np.random.rand(3 * len(self.atoms), 3 * len(self.atoms)) - 0.5
-        M[:, :6] = np.column_stack((t1, t2, t3, t4, t5, t6))
+        M = np.eye(3 * len(self.atoms))
+
+        i = 0
+        for t_i in (t1, t2, t3, t4, t5, t6):
+            if not np.isclose(np.linalg.norm(t_i), 0):
+                M[:, i] = t_i
+                i += 1
 
         return np.linalg.qr(M)[0]
```

Comment:
I am not sure what the QR factorization is doing here?

Reply:
it's to orthogonalise the projection matrix. Gram–Schmidt also works, and gives the same result from my testing, but is slower